tests/lib/rules/no-cond-assign.js

```diff
@@ -36,7 +36,10 @@ ruleTester.run("no-cond-assign", rule, {
         "while (someNode || (someNode = parentNode)) { }",
         "do { } while (someNode || (someNode = parentNode));",
         "for (;someNode || (someNode = parentNode););",
-        "if ((function(node) { return (node = parentNode); })(someNode)) { }",
+        { code: "if ((function(node) { return (node = parentNode); })(someNode)) { }", options: ["except-parens"] },
+        { code: "if ((function(node) { return (node = parentNode); })(someNode)) { }", options: ["always"] },
+        { code: "if ((node => node = parentNode)(someNode)) { }", options: ["except-parens"], parserOptions: { ecmaVersion: 6 } },
+        { code: "if ((node => node = parentNode)(someNode)) { }", options: ["always"], parserOptions: { ecmaVersion: 6 } },
```

Comment:
I'd love to see a test just for the function expression itself (not immediately invoked). Yes, that's a useless/constant condition, but it'd be good to make sure that case is covered anyway.